test/cypress/e2e/integration/app-frontend/attachments-in-group.js

```diff
@@ -0,0 +1,462 @@
+          continue;
+        }
+
+        out[key] = [];
+        for (const attachment of fullState.attachments.attachments[key]) {
+          out[key].push(attachment.name);
+        }
+      }
+
+      return selector ? selector(out) : out;
+    });
+  };
+
+  const simplifyFormData = (s) => {
+    // Find all attachment IDs and add them to a mapping so we can replace them in formData with their file names,
+    // since all our file names are unique anyway, and the UUIDs will change every time.
+    const idToNameMapping = {};
+    for (const attachmentList of Object.values(s.attachments.attachments)) {
+      for (const attachment of attachmentList) {
+        if (attachment.id && attachment.name) {
+          idToNameMapping[attachment.id] = attachment.name;
+        }
+      }
+    }
+
+    const expectedPrefix = 'Endringsmelding-grp-9786.OversiktOverEndringene-grp-9788';
+    return Object.keys(s.formData.formData)
+      .filter((key) => key.startsWith(expectedPrefix) && key.includes('fileUpload'))
+      .map((key) => {
+        const uuid = s.formData.formData[key];
+        if (idToNameMapping[uuid]) {
+          return [key.replace(expectedPrefix, ''), idToNameMapping[uuid]];
+        }
+
+        return [key.replace(expectedPrefix, ''), uuid];
+      })
+      .sort();
+  };
+
+  const interceptFormDataSave = () => {
+    cy.url().then((url) => {
+      const instanceId = url.split('/').slice(-2).join('/');
+      cy.intercept('PUT', `**/instances/${instanceId}/data/**`).as('saveInstanceData');
+    });
+
+  };
+
+  const waitForFormDataSave = () => {
+    cy.wait('@saveInstanceData').its('response.statusCode').should('eq', 201);
+  };
+
+  it('Works when uploading attachments to repeating groups, supports deleting attachments and entire rows', () => {
+    const filenames = [
+      {
+        single: 'singleFileInFirstRow.pdf',
+        multi: [
+          'multiInFirstRow1.pdf',
+          'multiInFirstRow2.pdf',
+          'multiInFirstRow3.pdf',
+        ],
+        nested: [
+          [
+            'nested-row0-sub0-1.pdf',
+            'nested-row0-sub0-2.pdf',
+            'nested-row0-sub0-3.pdf',
+          ],
+          [
+            'nested-row0-sub1-1.pdf',
+            'nested-row0-sub1-2.pdf',
+            'nested-row0-sub1-3.pdf',
+          ]
+        ],
+      },
+      {
+        single: 'singleFileInSecondRow.pdf',
+        multi: [
+          'multiInSecondRow1.pdf',
+          'multiInSecondRow2.pdf',
+          'multiInSecondRow3.pdf',
+          'multiInSecondRow4.pdf',
+        ],
+        nested: [
+          [
+            'nested-row1-sub0-1.pdf',
+            'nested-row1-sub0-2.pdf',
+            'nested-row1-sub0-3.pdf',
+          ],
+          [
+            'nested-row1-sub1-1.pdf',
+            'nested-row1-sub1-2.pdf',
+            'nested-row1-sub1-3.pdf',
+          ]
+        ],
+      }
+    ];
+
+    uploadFile(appFrontend.group.rows[0].uploadSingle, 0, filenames[0].single, true);
```

Comment:
I would really prefer if we used [RORO pattern](https://www.freecodecamp.org/news/elegant-patterns-in-modern-javascript-roro-be01e7669cbd/) here (and in general when accepting multiple arguments) - makes it so much easier to read and understand what the arguments are when calling the function. What does `0` and `true` mean here f.ex? We have to go to the definition to get the answers

Using RORO would also make it possible to call the function with `isTaggedUploader` set to true, without having to specify false for `verifyTableRow`.


```suggestion
    uploadFile({
	item: appFrontend.group.rows[0].uploadSingle, 
	attachmentIndex: 0, 
	fileName: filenames[0].single,
	verifyTableRow: true
    });
```

Reply:
Yup, I totally get it. In IntelliJ, hints on parameter names are built-in (so I rarely have the same need for this) but I rewrote to using RORO for this function now.